### PR TITLE
Convert tuple indices to arrays in NumpyArrayAdapter.ndarray and ArrayStack

### PR DESCRIPTION
**Pull request by pp-mo**
Stolen from https://github.com/djkirkham/biggus/tree/convert-tuple-indice-to-arrays

### CODE REVIEW

**Comment by marqh** [comment]
@pp-mo @djkirkham 
the test failures mean this does not appear to be a replacement right now.

Are these just unit test failures which would be expected during a re-implementation?
What would changes to the test look like to get them all passing?

File: biggus/_init.py [hunk 1]
```diff
@@ -804,7 +804,7 @@ def __getitem__(self, keys):
                     new_size = len(range(*key.indices(1)))
                     if new_size != 1:
                         broadcast_dict[key_index] = new_size
-                elif isinstance(key, tuple):
+                elif isinstance(key, tuple, np.ndarray):
```
Comment (djkirkham) [inline biggus/_init.py]: Change this line to:

`elif isinstance(key, (tuple, np.ndarray)):`

**Comment by djkirkham** [comment]
> Are these just unit test failures which would be expected during a re-implementation?

No, this is my fault. I've added a comment to the offending line of code with the suggested change.

**Comment by pp-mo** [comment]
This whole PR may or may not be wanted now ... : see https://github.com/SciTools/iris/issues/2120#issuecomment-243726523

**Comment by pp-mo** [comment]
> This whole PR may or may not be wanted now

In view of an internal requirement for ongoing numpy 1.8 compatibility, I think it ***is*** wanted.

After looking into the implications and talking through with @dkillick, I have eventually convinced myself that the apparent  kludge discussed at https://github.com/SciTools/biggus/pull/188/files#r76957286 (with a suitable code fix!) **is** in fact appropriate ...
This change merely removes the restriction on indexing with an array of indices : that restriction is *specific* to NewAxesArray : other biggus Array subtypes are perfectly ok with that usage (at least, most of them certainly are).

I will post here a fixed/amended version of this code when I've got it sorted out...

**Comment by pp-mo** [comment]
> I will post here a fixed/amended version of this code when I've got it sorted out...

I changed my mind:  Since wondering whether any other classes might also need a similar tweak, I think I have now found more general fix.
Replaced by : #189

**Pull request closed without merging**